shared/teams/team/rows/channel-row/channel.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react'
+import * as Types from '../../../../constants/types/teams'
+import * as ChatTypes from '../../../../constants/types/chat2'
+import * as Constants from '../../../../constants/teams'
+import * as Kb from '../../../../common-adapters'
+import * as Container from '../../../../util/container'
+import * as TeamsGen from '../../../../actions/teams-gen'
+import * as Styles from '../../../../styles'
+import {Activity} from '../../../common'
+import {pluralize} from '../../../../util/string'
+
+type ChannelRowProps = {
+  channel: Types.ChannelInfo
+  teamID: Types.TeamID
+  conversationIDKey: ChatTypes.ConversationIDKey
+}
+const ChannelRow = (props: ChannelRowProps) => {
+  const {channel, teamID, conversationIDKey} = props
+  const isGeneral = channel.channelname === 'general'
+
+  const _selected = Container.useSelector(
+    state => !!state.teams.selectedChannels.get(teamID)?.has(channel.channelname)
+  )
+  const [selected, setSelected] = React.useState(_selected)
```

Comment:
why is this managing this state locally? the store can be the source of truth for this.